src/apps/vpn/mozillavpn.cpp

```diff
@@ -889,17 +889,12 @@ void MozillaVPN::logout() {
   }
 
   if (m_private->m_deviceModel.hasCurrentDevice(keys())) {
-    TaskScheduler::scheduleTask(new TaskGroup(
-        {new TaskRemoveDevice(keys()->publicKey()),
-         // Immediately after the scheduling of the device removal, we want to
-         // delete the session token, so that, in case the app is terminated, at
-         // the next execution we go back to the init screen.
-         new TaskFunction([this]() { reset(false); })}));
-
-    // In case the app is closed even before scheduling the previous TaskGroup,
-    // removing the key we will enforce a new authentication at the first
-    // TaskAccount execution.
-    m_private->m_keys.forgetKeys();
+    TaskScheduler::scheduleTask(new TaskRemoveDevice(keys()->publicKey()));
+
+    // Immediately after the scheduling of the device removal, we want to
+    // delete the session token, so that, in case the app is terminated, at
+    // the next execution we go back to the init screen.
+    reset(false);
```

Comment:
Are you changing this just because reset is now possible because the task copied the auth header and there is not need to run them in sequence?

Reply:
Correct! Before this was not possible because of how NetworkRequest was written: the authorization header was computer in the CTOR. But now, the caller has full control of the headers and the authorization one can be set with a cached string value.